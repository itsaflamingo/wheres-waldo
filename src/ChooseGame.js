import React, { useState } from "react";
import './styles/SetGame.css';
import AliceCarousel from 'react-alice-carousel';
import "react-alice-carousel/lib/alice-carousel.css";
// import SetGame from "./components/SetGame.js";
import MakeBackgrounds from "./components/Backgrounds";
import ChooseGameDiv from "./components/ChooseGameDiv";

function ChooseGame() {

    const [showGame, setShowGame] = useState(false);
    const [game, SetGame] = useState({});
    const [games, setGames] = useState(MakeBackgrounds());

    const startGame = (e) => {
        setShowGame(true)
    }

    return(
        <div id='page'>
                <ChooseGameDiv games={games} />
                <div id='startBtn'>
                    <button id='start'
                    onClick={(e) => startGame(e)}>Start</button>
            </div>
            {/* {showGame && <SetGame game={game} />} */}
        </div>
    )
}

export default ChooseGame;