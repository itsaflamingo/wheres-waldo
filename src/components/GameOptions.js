import React from "react";
import GameOption from "./GameOption";

function GameOptions(props) {

    const {games} = props;

    return (
        <div id='choose-game'>
        {games.map((game) => {
            return(
                <div className="game" 
                    id={game.name} 
                    key={game.id}>
                    <GameOption game={game} />
                </div>
            )
        })}
        
        </div>
    )
}

export default GameOptions;