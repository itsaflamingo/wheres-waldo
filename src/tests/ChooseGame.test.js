import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";  // optional
import {BrowserRouter as Router} from 'react-router-dom';
import ChooseGame from '../ChooseGame'
import GameOptions from '../components/GameOptions';
import MakeBackgrounds from '../components/Backgrounds'
import { act } from "react-dom/test-utils";
import userEvent from "@testing-library/user-event";

describe('Choose game component', () => {

    const games = MakeBackgrounds(); 
    const setGame = jest.fn();
    const game=games[0];    
    it('Renders', () => {
        const {container} = render(
            <Router>
                <ChooseGame game={game} setGame={setGame} />
            </Router>
        )
        expect(container).toMatchSnapshot();
    })
    it('Renders headings correctly', () => {

        render(
            <Router>
                <ChooseGame game={game} setGame={setGame} />
            </Router>
        )

        expect(screen.getAllByRole('heading')[0].textContent).toMatch('A.D.2222')
        expect(screen.getAllByRole('heading')[1].textContent).toMatch('Robot City')
        expect(screen.getAllByRole('heading')[2].textContent).toMatch('Universe 113')
    })
    it('Renders images correctly', () => {
        render(<GameOptions games={games} />)

        expect(screen.getAllByTestId('background')[0]).toHaveStyle(`background-image: url(${games[0].image})`)
        expect(screen.getAllByTestId('background')[1]).toHaveStyle(`background-image: url(${games[1].image})`)
        expect(screen.getAllByTestId('background')[2]).toHaveStyle(`background-image: url(${games[2].image})`)

        expect(screen.getAllByTestId('character')[0]).toHaveStyle(`background-image: url(${games[0].characters[0].image})`)
        expect(screen.getAllByTestId('character')[1]).toHaveStyle(`background-image: url(${games[0].characters[1].image})`)
        expect(screen.getAllByTestId('character')[2]).toHaveStyle(`background-image: url(${games[0].characters[2].image})`)

        expect(screen.getAllByTestId('character')[3]).toHaveStyle(`background-image: url(${games[1].characters[0].image})`)
        expect(screen.getAllByTestId('character')[4]).toHaveStyle(`background-image: url(${games[1].characters[1].image})`)
        expect(screen.getAllByTestId('character')[5]).toHaveStyle(`background-image: url(${games[1].characters[2].image})`)

        expect(screen.getAllByTestId('character')[6]).toHaveStyle(`background-image: url(${games[2].characters[0].image})`)
        expect(screen.getAllByTestId('character')[7]).toHaveStyle(`background-image: url(${games[2].characters[1].image})`)
        expect(screen.getAllByTestId('character')[8]).toHaveStyle(`background-image: url(${games[2].characters[2].image})`)
    })

    it('When clicked, button navigates to new page', () => {
        render(
            <Router>
                <ChooseGame game={game} setGame={setGame} />
            </Router>)

        const button = screen.getByRole('button', {name: 'Start'});
        const chosenGame = screen.getByText(/Robot City/i);

        act(() => {
            userEvent.click(chosenGame);
            button.dispatchEvent(new MouseEvent('click', {bubbles: true}))
        });
        expect(window.location.pathname).toEqual('/play');
    })
})