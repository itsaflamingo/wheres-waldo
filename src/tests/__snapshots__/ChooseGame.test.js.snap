// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choose game component Renders 1`] = `
<div>
  <div
    id="page"
  >
    <div
      id="info-container"
    >
      <button
        id="show-info"
      >
        Information
      </button>
    </div>
    <div
      id="choose-game"
    >
      <div
        class="game"
        id="A.D.2222"
      >
        <div
          class="game-container"
        >
          <h2
            class="title"
          >
            A.D.2222
          </h2>
          <div
            class="content"
          >
            <div
              class="background"
              data-testid="background"
              style="background-image: url(img1.jpg); background-size: cover; background-repeat: no-repeat;"
            />
            <div
              class="characters"
            >
              <div
                class="character-div"
              >
                <div
                  class="character"
                  data-testid="character"
                  style="background-image: url(hollow-knight.webp); background-size: contain; background-repeat: no-repeat;"
                />
                <div
                  class="name"
                >
                  Hollow Knight
                </div>
              </div>
              <div
                class="character-div"
              >
                <div
                  class="character"
                  data-testid="character"
                  style="background-image: url(yubaba.png); background-size: contain; background-repeat: no-repeat;"
                />
                <div
                  class="name"
                >
                  Yubaba
                </div>
              </div>
              <div
                class="character-div"
              >
                <div
                  class="character"
                  data-testid="character"
                  style="background-image: url(bowser.png); background-size: contain; background-repeat: no-repeat;"
                />
                <div
                  class="name"
                >
                  Bowser
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="game"
        id="Robot City"
      >
        <div
          class="game-container"
        >
          <h2
            class="title"
          >
            Robot City
          </h2>
          <div
            class="content"
          >
            <div
              class="background"
              data-testid="background"
              style="background-image: url(robotcity.jpg); background-size: cover; background-repeat: no-repeat;"
            />
            <div
              class="characters"
            >
              <div
                class="character-div"
              >
                <div
                  class="character"
                  data-testid="character"
                  style="background-image: url(meg.webp); background-size: contain; background-repeat: no-repeat;"
                />
                <div
                  class="name"
                >
                  Meg
                </div>
              </div>
              <div
                class="character-div"
              >
                <div
                  class="character"
                  data-testid="character"
                  style="background-image: url(goku.webp); background-size: contain; background-repeat: no-repeat;"
                />
                <div
                  class="name"
                >
                  Goku
                </div>
              </div>
              <div
                class="character-div"
              >
                <div
                  class="character"
                  data-testid="character"
                  style="background-image: url(beast-titan.png); background-size: contain; background-repeat: no-repeat;"
                />
                <div
                  class="name"
                >
                  Beast Titan
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="game"
        id="Universe 113"
      >
        <div
          class="game-container"
        >
          <h2
            class="title"
          >
            Universe 113
          </h2>
          <div
            class="content"
          >
            <div
              class="background"
              data-testid="background"
              style="background-image: url(img2.jpg); background-size: cover; background-repeat: no-repeat;"
            />
            <div
              class="characters"
            >
              <div
                class="character-div"
              >
                <div
                  class="character"
                  data-testid="character"
                  style="background-image: url(stewie.png); background-size: contain; background-repeat: no-repeat;"
                />
                <div
                  class="name"
                >
                  Stewie
                </div>
              </div>
              <div
                class="character-div"
              >
                <div
                  class="character"
                  data-testid="character"
                  style="background-image: url(bender.png); background-size: contain; background-repeat: no-repeat;"
                />
                <div
                  class="name"
                >
                  Bender
                </div>
              </div>
              <div
                class="character-div"
              >
                <div
                  class="character"
                  data-testid="character"
                  style="background-image: url(totoro.png); background-size: contain; background-repeat: no-repeat;"
                />
                <div
                  class="name"
                >
                  Totoro
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      id="start-btn"
    >
      <button
        id="start"
      >
        Start
      </button>
    </div>
  </div>
</div>
`;
